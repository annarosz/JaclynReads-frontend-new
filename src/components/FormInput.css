.form-group {
  display: flex;
  flex-direction: column;
  /* background-color: pink; */
}

.form-control {
  /* background-color: white; */
  min-width: 200px;
  max-width: 600px;
  height: 50px;
  border: 0px;
  outline: none;
  padding: 10px;
  /* top /right / bottom / left */
  margin: 5px 0px 15px 0px;
}

.input {
  background-color: lavender;
}

input[type=text] {
  background-color: white;
  /* border: solid 0.5pt #138B8A; */
  /* border-radius: 30px; */
}

.searchBorder {
  min-width: 180px;
  border: solid 1px #138B8A;
  padding: 10px;
  border-radius: 30px;
}

#publisher {
  width: 250px;
  height: 50px;
}

#year {
  width: 250px;
  height: 50px;
}