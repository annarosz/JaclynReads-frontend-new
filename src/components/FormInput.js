import React from 'react';
import './FormInput.css';

const FormInput = props => {
  return (
    <>
    
    <div className="form-group">
      <label htmlFor={props.name} className="form-label">{props.title}</label>
      <input
        className="form-control"
        id={props.name}
        name={props.name}
        type={props.inputType}
        value={props.value}
        onChange={props.handleChange}
        placeholder={props.placeholder}
        min='1999'
        noValidate
      />
    </div>
    </>
  );
};

export default FormInput;