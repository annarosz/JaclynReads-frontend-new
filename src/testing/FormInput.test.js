import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";
import FormInput from '../components/FormInput'

describe("<FormInput />", () => {

  const reviewList = [
    {
      name: "Three Little Pigs",
      author: "Big Bad Wolf",
      id: "5d19cd74a0c3e2c7f68ff800",
      inputType: "type",
      value: "value",
      handleChange: "handleChange"
      
    }
  ]


  it("displays loading when no review in list", () => {
    const review = [];
    const wrapper = shallow(<FormInput />);
    expect(wrapper).toMatchSnapshot();
  });


  it("props are displayed in input fields", () => {
  
    const wrapper = shallow(<FormInput 
      id={reviewList[0].name}
      name={reviewList[0].name}
      inputType={reviewList[0].inputType}
      value={reviewList[0].value}
      handleChange={reviewList[0].handleChange}
      />);
    expect(wrapper).toMatchSnapshot();
  });



})