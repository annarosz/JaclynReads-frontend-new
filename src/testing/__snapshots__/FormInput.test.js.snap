// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> displays loading when no review in list 1`] = `
<Fragment>
  <div
    className="form-group"
  >
    <label
      className="form-label"
    />
    <input
      className="form-control"
      min="1999"
      noValidate={true}
    />
  </div>
</Fragment>
`;

exports[`<FormInput /> props are displayed in input fields 1`] = `
<Fragment>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="Three Little Pigs"
    />
    <input
      className="form-control"
      id="Three Little Pigs"
      min="1999"
      name="Three Little Pigs"
      noValidate={true}
      onChange="handleChange"
      type="type"
      value="value"
    />
  </div>
</Fragment>
`;
